config_parser_test.cc

```diff
@@ -35,14 +36,23 @@ TEST_F(NginxStringConfigTest, AnotherSimpleConfig) {
 	EXPECT_TRUE(ParseString("foo bar;"));
 	EXPECT_EQ(1, out_config_.statements_.size())
 		<< "Config has one statements";
-	EXPECT_EQ("foo", out_config_.statements_.at(0)->tokens_.at(0));
+	EXPECT_EQ("foo", out_config_.statements_[0]->tokens_[0]);
+	EXPECT_EQ("bar", out_config_.statements_[0]->tokens_[1]);
 }
 
-TEST_F(NginxStringConfigTest, InvalidConfig) {
+TEST_F(NginxStringConfigTest, EmptyConfig) {
+    EXPECT_FALSE(ParseString(";"));
+    EXPECT_FALSE(ParseString(""));
+	EXPECT_FALSE(ParseString("server { }"));
+}
+
+TEST_F(NginxStringConfigTest, InvalidConfigStatement) {
 	EXPECT_FALSE(ParseString("foo bar"));
+	EXPECT_FALSE(ParseString("server { listen 80 }"));
 }
 
-TEST_F(NginxStringConfigTest, NestedConfig) {
-	EXPECT_TRUE(ParseString("server { listen 80; }"));
-	// TODO: Test the contents of out_config_;
+TEST_F(NginxStringConfigTest, NestedConfigStatement) {
+	EXPECT_FALSE(ParseString("server { listen 80; "));
+	EXPECT_EQ(1, out_config_.statements_.size()) << "Config has one statement";
```

Comment:
I don't think you should be testing out_config_ for a case where your config was not parsed properly (i.e parse function returned false). But out_config_ should definitely be tested, so that's a good thought. Maybe change this test so that your parser returns true?